### PR TITLE
Create models

### PR DESCRIPTION
**Pull request by Abel-desu**
- Create models
- Use models to insert data
- Use other [CRUD methods](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/active_record_crud.md) to find, update and delete entities.
- Create custom methods

### CODE REVIEW

File: app/models/user.rb [hunk 1]
```diff
@@ -1,2 +1,5 @@
 class User < ApplicationRecord
+  has_many :posts, foreign_key: :author_id, dependent: :destroy
+  has_many :comments, foreign_key: :author_id, dependent: :destroy
+  has_many :likes, foreign_key: :author_id, dependent: :destroy
 end
```
Comment (tufoinnkuo10) [inline app/models/user.rb:9]: - [ ] please kindly add in the user model a method that returns 3 most recent posts
eg 
```ruby
def recent_post
  posts.order('created_at Desc').limit(3)
end
```

File: app/models/post.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments, foreign_key: :post_id, dependent: :destroy
+  has_many :likes, foreign_key: :post_id, dependent: :destroy
+end
```
Comment (tufoinnkuo10) [inline app/models/post.rb:15]: - [ ] please add to the post model a method that updates the posts counter for a user

File: app/models/post.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class Post < ApplicationRecord
+  belongs_to :author, class_name: 'User'
+  has_many :comments, foreign_key: :post_id, dependent: :destroy
+  has_many :likes, foreign_key: :post_id, dependent: :destroy
```
Comment (tufoinnkuo10) [inline app/models/post.rb:4]: - [ ] please add to the post model a method that returns the 5 most recent comments for a given post
eg
```ruby
def recent_comments
  comments.order('created_at Desc').limit(5)
end
```

File: app/models/comment.rb [hunk 4]
```diff
@@ -1,6 +1,4 @@
 class Comment < ApplicationRecord
-  belongs_to :AuthorId
-  belongs_to :PostId
-  belongs_to :UpdateAT
-  belongs_to :CreateAT
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
 end
```
Comment (tufoinnkuo10) [inline app/models/comment.rb:10]: - [ ] kindly ensure the comment model has a method that updates the comments counter for a post

File: app/models/like.rb [hunk 5]
```diff
@@ -1,4 +1,4 @@
 class Like < ApplicationRecord
-  belongs_to :CreatedAT
-  belongs_to :UpdateAt
+  belongs_to :author, class_name: 'User'
+  belongs_to :post
```
Comment (tufoinnkuo10) [inline app/models/like.rb:3]: - [ ] please add to the like model a method that updates the likes counter for a post
eg
```ruby
def update_likes_counter
  post.increment!(:likes_counter)
end
```

**Review by tufoinnkuo10 — CHANGES_REQUESTED** [review]
## Hi @Abel-desu 

There are some issues that you still need to work on to go to the next project but you are almost there!

To Highlight
✅ You used the correct Github flow
✅ Descriptive pull request message
✅ linter error checks validated
✅ Documented your work professionally


## Required Changes ♻️
Check the comments under the review
Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

File: app/models/like.rb [hunk 6]
```diff
@@ -1,4 +1,4 @@
 class Like < ApplicationRecord
-  belongs_to :CreatedAT
-  belongs_to :UpdateAt
+  belongs_to :author, class_name: 'User'
+  belongs_to :post, counter_cache: :likesCounter
 end
```
Comment (AnselemOdims) [inline app/models/like.rb]: - [ ] As pointed out by the previous reviewer, please kindly include a method that updates the likes counter.

**Review by AnselemOdims — CHANGES_REQUESTED** [review]
### Hi @Abel-desu , 
Good job implementing some of the changes requested by the previous reviewer.

## Required Changes ♻️
- [ ] Please also check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity! You can tag me using @AnselemOdims for a quicker response

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Review by AnselemOdims — APPROVED** [review]
### Hi  @Abel-desu 🙋‍♀️, 
Good job in making the required changes 👏 

# STATUS APPROVED ✔
Your project is complete!  There is nothing else to say other than... it's time to merge it 🔗 🚢 
Congratulations! 🎉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers 🥂  and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Use the tag @AnselemOdims for a quicker reply


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**